src/test/java/com/googleinterns/zoomtube/servlets/LectureServletTest.java

```diff
@@ -83,11 +82,40 @@ public void tearDown() {
     testServices.tearDown();
   }
 
+  @Test
+  public void doPost_missingName_badRequest() throws IOException, ServletException {
+    when(request.getParameter(LectureServlet.PARAM_LINK)).thenReturn(TEST_LINK);
+
+    servlet.doPost(request, response);
+
+    verify(response).sendError(HttpServletResponse.SC_BAD_REQUEST, LectureServlet.ERROR_MISSING_NAME);
+  }
+
+  @Test
+  public void doPost_missingLink_badRequest() throws IOException, ServletException {
+    when(request.getParameter(LectureServlet.PARAM_NAME)).thenReturn(TEST_NAME);
+
+    servlet.doPost(request, response);
+
+    verify(response).sendError(HttpServletResponse.SC_BAD_REQUEST, LectureServlet.ERROR_MISSING_LINK);
+  }
+
+  @Test
+  public void doPost_invalidLink_badRequest() throws IOException, ServletException {
+    when(request.getParameter(LectureServlet.PARAM_NAME)).thenReturn(TEST_NAME);
+    when(request.getParameter(LectureServlet.PARAM_LINK)).thenReturn("this is not a link");
+
+    servlet.doPost(request, response);
+
+    verify(response).sendError(HttpServletResponse.SC_BAD_REQUEST, LectureServlet.ERROR_INVALID_LINK);
+  }
+
   @Test
   public void doPost_urlAlreadyInDatabase_shouldReturnLecture()
       throws IOException, ServletException {
-    when(request.getParameter(LINK_INPUT)).thenReturn(TEST_LINK);
-    datastoreService.put(LectureUtil.createEntity(/* lectureName= */ "", TEST_LINK, TEST_ID));
+    when(request.getParameter(LectureServlet.PARAM_NAME)).thenReturn(TEST_NAME);
+    when(request.getParameter(LectureServlet.PARAM_LINK)).thenReturn(TEST_LINK);
+    datastoreService.put(LectureUtil.createEntity(TEST_NAME, TEST_LINK, TEST_ID));
```

Comment:
TEST does not add any value to what the variable is. I believe better names would be:

TEST_NAME => LECTURE_NAME
TEST_LINK => URL_LINK
TEST_ID => LECTURE_ID

What do you think?

Reply:
This PR did not create those constants, so I'm not going to change them in this PR.  Added tracking issue #216

Reply:
sgtm